### DISCUSSION TITLE
Will DinD mode in runner scale set persist over time and be supported by Github ?

### DISCUSSION THREAD

**Discussion by giom-l** [actions/actions-runner-controller#3237] [Questions]
Hello, 

In our organization, we have a difference of understandings about future of self-hosted runners.
We were previously using Actions Runner Controller with old images and recently switched to runner scale set. 

By reading [this discussion](https://github.com/actions/actions-runner-controller/discussions/2775), we took note about what will be still be supported by Github and what will be left to community. 

Maintainers of our self-hosted runners chose the `kubernetes` deployment type, which prevent us to use any `docker` command

The point on which we don't all agree on our side is about the future support of DinD mode. 
[I know the runner scale set currently supports both `kubernetes` and `dind` modes](https://github.com/actions/actions-runner-controller/blob/47dfed3ced786c117a58cbacedd8b5fb90777882/charts/gha-runner-scale-set/values.yaml#L79)

But some of us think it is only supported for some time by Github and will be only maintained by community in the future.

Would it be possible to have a clear point of view about this ? 
Will the `dind` mode support get removed eventually ?

**Comment by Link-** [comment]
We do not have plans to remove `dind` mode. Both modes exist because there are tradeoffs to be made using either of them and you will have to choose one that best fit your environment. 

With `dind` you'll have to run your containers in privileged mode even when you use `dind-rootless`. With `kubernetes` mode you cannot run Docker command but you can use Kaniko for building containers for example.

**Reply by giom-l** [reply]
Thanks a lot for your reply @Link- 
So both modes will continue to receive features/fixes and so on since those are complementary ?

**Reply by daoz1026** [reply]
Hello, we would like to know also, if the `dind` mode will be supported as well as the `kubernetes` mode. Will they both receive features/fixes ? 
Regards

**Reply by Link-** [reply]
Both are supported for now
